gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/access/GemFireTransaction.java

```diff
@@ -2918,8 +2918,12 @@ public final void beginTransaction(final IsolationLevel isolationLevel,
       final TXStateInterface gemfireTx = TXManagerImpl.getCurrentSnapshotTXState();
 
       if (tx != null && tx != TXStateProxy.TX_NOT_SET && gemfireTx != tx) {
-        this.txManager.commit(tx, this.connectionID, TXManagerImpl.FULL_COMMIT,
-            null, false);
+        // don't commit as it may have reference to other one whereas gemfire tx is
+        // going to be used.
+        if (gemfireTx == null) {
```

Comment:
Would you want to add additional "snappydata" mode in the check if this is specific to SnappyData and not gemfirexd.